anomaly_detectors/anomalib_lmi/anomaly_model.py

```diff
@@ -1,178 +1,238 @@
-                        self.fp16 = True
-                else:
-                    self.output_names.append(name)
-                im = self.from_numpy(np.empty(shape, dtype=dtype)).to(self.device)
-                self.bindings[name] = Binding(name, dtype, shape, im, int(im.data_ptr()))
-            self.binding_addrs = OrderedDict((n, d.ptr) for n, d in self.bindings.items())
-            self.model_shape=list(shape[-2:])
-            self.inference_mode='TRT'
-        elif ext=='.pt':     
+        
+        self.version = version
+        self.loaded = False
+        try:
             model = torch.load(model_path,map_location=self.device)["model"]
             model.eval()
             self.pt_model=model.to(self.device)
             self.pt_metadata = torch.load(model_path, map_location=self.device)["metadata"] if model_path else {}
-            self.pt_transform=A.from_dict(self.pt_metadata["transform"])
-            for d in self.pt_metadata['transform']['transform']['transforms']:
-                if d['__class_fullname__']=='Resize':
-                    self.model_shape = [d['height'], d['width']]
+            self.logger.info(f"Model metadata: {self.pt_metadata}")
+            if "transform" in self.pt_metadata:
+                self.pt_transform=A.from_dict(self.pt_metadata["transform"])
+                self.version = 'v0'
+            else:
+                self.pt_transform = None
+                self.version = 'v1'
+            
+            if self.version != version:
+                self.logger.warning(f"Model version updated {self.version}")
+
+            if self.version == 'v0':
+                for d in self.pt_metadata['transform']['transform']['transforms']:
+                    if d['__class_fullname__']=='Resize':
+                        self.model_shape = [d['height'], d['width']]
+            elif self.version == 'v1':
+                for d in self.pt_model.transform.transforms:
+                    if isinstance(d, v2.Resize):
+                        self.model_shape = to_list(d.size)
             self.inference_mode='PT'
-        else:
-            raise Exception(f'Unknown model format: {ext}')
-        
-        
-    @torch.inference_mode()
-    def normalize(self,image: np.ndarray) -> np.ndarray:
-        """
-        Desc: Normalize the image to the given mean and standard deviation for consistency with pytorch backbone
-        """
-        image = image.astype(np.float32)
-        mean = (0.485, 0.456, 0.406)
-        std = (0.229, 0.224, 0.225)
-        image /= 255.0
-        image -= mean
-        image /= std
-        return image
-    
-    
-    @torch.inference_mode()
-    def preprocess(self, image):
-        '''
-        Desc: Preprocess input image.
-        args:
-            - image: numpy array [H,W,Ch]
+            self.loaded = True
+        except Exception as e:
+            self.logger.exception(
+                f"Unable to load pt model using v0 and v1 {e}")
```

Comment:
This would generate error msgs if loading a pytorch trace model. The user might think there is a bug in the code. How about generate error msgs if the pt file cannot be loaded using both approaches (original pt and traced/scripted pt)?